Add about section

MathiasLantean opened the pull request:
Extended texts are likely to change, this API was created for letting the user edit static texts with no deploy required.

Commit:
Add base app for contact

This app manages all general info the user needs to see (about section, terms of usage, etc.), as well as chats and connection.

Commit:
Add Information model

This model will save long texts that are likely to change, so admins can change them from the Django admin.
Each Info section will have only one instance and it must be created through data migrations, Django admin only allows admins to edit that instance (neither create nor delete).

Commit:
Test refactor

As Factory has a method that creates a batch of objects the for sentence is no needed.

Commit:
Add API for retrieving general information

This API is used for getting text and system information that should be stored in the DB in order to make them easier to change by admins.

Commit:
Add test for Information

guilleijo commented on contact/views.py:
```diff
@@ -0,0 +1,19 @@
+        except Information.DoesNotExist:
+            raise Http404
+
+    def get(self, request, pk):
+        info = self.get_object(pk)
+        serializer = InformationSerializer(info)
+        return Response(serializer.data)
```
I think you can remove the `get_object` method and use `info = get_object_or_404(Information, pk=pk)`. You need to import it, check the [example](https://docs.djangoproject.com/en/3.0/topics/http/shortcuts/#get-object-or-404).

Also, have you considered using a simple viewset? Maybe `ReadOnlyModelViewSet` if you only want to list and get objects.

MathiasLantean replied:
Thanks for the `get_object_or_404`, I totally forgot about it. 

Using `ReadOnlyModelViewSet` could be an option but as I don't want that the user can list all the info stored in the DB, an APIView just with the get method defined fits perfectly.

MathiasLantean replied:
Okay, finally I decided to replace `APIView` for  `GenericViewSet, RetrieveModelMixin`. It does the same but with less code and using features of ViewSet. As Guillermo explains to me, this allows me to extend the behavior in case the API needs changes in the future.

guilleijo commented on contact/migrations/0002_initial_contact_info.py:
```diff
@@ -0,0 +1,24 @@
+    Information.objects.create(title='about', detail='')
+
+
+def add_remove_info(apps, schema_editor):
```
Shouldn't this be `remove_about_info`?

guilleijo reviewed (commented):
Left some comments

Commit:
Code refactor

Rename function in data migration. 
Change the URL name in tests as router.register is used instead of path.
Rewrite API view using GenericViewSet and RetriveModelMixin.

guilleijo reviewed (approved):
👍
